corehq/apps/userreports/dbaccessors.py

```diff
@@ -183,124 +181,115 @@ def delete_all_ucr_tables_for_domain(domain):
-        except KeyError:
-            print(f"Engine id {engine_id} does not exist anymore. Skipping.")
-            continue
+def _get_existing_ucrs(engine_id, domain, ignore_active_domains):
+    try:
+        engine = connection_manager.get_engine(engine_id)
+    except KeyError:
+        raise ValueError(f"Engine id {engine_id} does not exist.")
+
+    with engine.begin() as connection:
+        # Using string formatting rather than execute with %s syntax
+        # is acceptable here because the strings we're inserting are static
+        # and only templated for DRYness
+        results = connection.execute(f"""
+        SELECT table_name
+          FROM information_schema.tables
+        WHERE table_schema='public'
+          AND table_type='BASE TABLE'
+          AND (
+            table_name LIKE '{UCR_TABLE_PREFIX}%%'
+            OR
+            table_name LIKE '{LEGACY_UCR_TABLE_PREFIX}%%'
+        );
+        """).fetchall()
+        all_existing_ucrs = [r[0] for r in results]
+
+    if domain:
+        all_existing_ucrs = [ucr for ucr in all_existing_ucrs if
+                             get_domain_for_ucr_table_name(ucr) == domain]
+    elif ignore_active_domains:
+        # can use elif because this is redundant after a domain filter
+        all_existing_ucrs = [ucr for ucr in all_existing_ucrs if
+                             Domain.is_domain_deleted(
+                                 get_domain_for_ucr_table_name(ucr))]
+
+    return all_existing_ucrs
+
+
+def _get_confirmed_orphaned_ucrs(engine_id, suspected_orphaned_ucrs):
+    if not suspected_orphaned_ucrs:
+        return []
+
+    engine = connection_manager.get_engine(engine_id)
+    confirmed_orphaned_ucrs = []
+    for suspected_orphaned_ucr in suspected_orphaned_ucrs:
         with engine.begin() as connection:
-            # Using string formatting rather than execute with %s syntax
-            # is acceptable here because the strings we're inserting are static
-            # and only templated for DRYness
-            results = connection.execute(f"""
-            SELECT table_name
-              FROM information_schema.tables
-            WHERE table_schema='public'
-              AND table_type='BASE TABLE'
-              AND (
-                table_name LIKE '{UCR_TABLE_PREFIX}%%'
-                OR
-                table_name LIKE '{LEGACY_UCR_TABLE_PREFIX}%%'
-            );
-            """).fetchall()
-            tables_in_db = {r[0] for r in results}
-        tables_without_data_sources[engine_id] = tables_in_db - expected_tables
-    return tables_without_data_sources
-
-
-def drop_ucr_tables(ucr_tables_by_engine_id):
+            try:
+                connection.execute(
+                    f'SELECT COUNT(*), MAX(inserted_at) FROM "{suspected_orphaned_ucr}"')
+            except ProgrammingError:
+                print(f"\t{suspected_orphaned_ucr}: no inserted_at column, probably not UCR")
```

Comment:
I think there are reasons other than "no inserted_at column" that could cause `ProgrammingError`. Does it matter if the error message is wrong?

Reply:
74dea6428e20b7b7146d9ef0064a8559a1024d28

Reply:
I'm not sure how much it matters and I don't have much context on why this message exists in the first place, but updated the wording a bit to be more clear.